src/log_surgeon/finite_automata/RegexDFA.hpp

```diff
@@ -50,6 +50,42 @@ class RegexDFAState {
     std::conditional_t<stateType == RegexDFAStateType::UTF8, Tree, std::tuple<>> m_tree_transitions;
 };
 
+template <typename DFAState>
+class RegexDFAStatePair {
+public:
+    /// TODO: a lot of this pointer usage could probably be references
+    RegexDFAStatePair(DFAState const* state1, DFAState const* state2)
+            : m_state1(state1),
+              m_state2(state2){};
+
+    bool operator<(RegexDFAStatePair const& rhs) const {
+        if (m_state1 == rhs.m_state1) {
+            return m_state2 < rhs.m_state2;
+        }
+        return m_state1 < rhs.m_state1;  // assume that you compare the record based on a
+    }
+
+    /**
+     * Generates all pairs reachable from the current pair via a single input.
+     * @return vector of reachable pairs
+     */
+    auto generate_reachable_pairs() -> std::set<RegexDFAStatePair>;
```

Comment:
```suggestion
    auto get_reachable_pairs() -> std::set<RegexDFAStatePair>;
```
Or do you feel like that's misleading?